src/server/services/wms/qgswmsparameters.cpp

```diff
@@ -1417,6 +1417,285 @@ namespace QgsWms
+  }
+
+  QList<QgsGeometry> QgsWmsParameters::composerMapParamToGeomList( int mapId, QString name, char delimiter ) const
+  {
+    QStringList l = composerMapParamToStringList( mapId, name, delimiter );
+    QList<QgsGeometry> geometries;
+
+    Q_FOREACH ( QString wkt, l )
+    {
+      QgsGeometry g( QgsGeometry::fromWkt( wkt ) );
+
+      if ( g.isGeosValid() )
+      {
+        geometries.append( g );
+      }
+      else
+      {
+        QString pMapId = "MAP" + QString::number( mapId );
+        QString paramName = pMapId + ":" + name;
+        QString valStr = l.join( delimiter );
+        QString msg = paramName + " ('" + valStr + "') cannot be converted into a list of geometries";
+        raiseError( msg );
+      }
+    }
+
+    return geometries;
+  }
+
+  QList<QColor> QgsWmsParameters::composerMapParamToColorList( int mapId, QString name, char delimiter ) const
+  {
+    QStringList l = composerMapParamToStringList( mapId, name, delimiter );
+    QList<QColor> elements;
+
+    Q_FOREACH ( QString element, l )
+    {
+      QColor c = QColor( element );
+
+      if ( c.isValid() )
+      {
+        elements.append( c );
+      }
+      else
+      {
+        QString pMapId = "MAP" + QString::number( mapId );
+        QString paramName = pMapId + ":" + name;
+        QString valStr = l.join( delimiter );
+        QString msg = paramName + " ('" + valStr + "') cannot be converted into a list of colors";
+        raiseError( msg );
+      }
+    }
+
+    return elements;
+  }
+
+  QList<int> QgsWmsParameters::composerMapParamToIntList( int mapId, QString name, char delimiter ) const
+  {
+    QStringList l = composerMapParamToStringList( mapId, name, delimiter );
+    QList<int> elements;
+
+    Q_FOREACH ( QString element, l )
+    {
+      bool ok;
+      int e = element.toInt( &ok );
+
+      if ( ok )
+      {
+        elements.append( e );
+      }
+      else
+      {
+        QString pMapId = "MAP" + QString::number( mapId );
+        QString paramName = pMapId + ":" + name;
+        QString valStr = l.join( delimiter );
+        QString msg = paramName + " ('" + valStr + "') cannot be converted into a list of int";
+        raiseError( msg );
+      }
+    }
+
+    return elements;
+  }
+
+  QList<float> QgsWmsParameters::composerMapParamToFloatList( int mapId, QString name, char delimiter ) const
+  {
+    QStringList l = composerMapParamToStringList( mapId, name, delimiter );
+    QList<float> elements;
+
+    Q_FOREACH ( QString element, l )
+    {
+      bool ok;
+      float e = element.toFloat( &ok );
+
+      if ( ok )
+      {
+        elements.append( e );
+      }
+      else
+      {
+        QString pMapId = "MAP" + QString::number( mapId );
+        QString paramName = pMapId + ":" + name;
+        QString valStr = l.join( delimiter );
+        QString msg = paramName + " ('" + valStr + "') cannot be converted into a list of float";
+        raiseError( msg );
+      }
+    }
+
+    return elements;
+  }
+
```

Comment:
Do you think we could use basic method `toXXX` in some way in these new methods `composerMapParamToXXX` instead of rewriting everything (conversion and exception raising)?